src/lib/utils/memory_resource_manager.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <tbb/concurrent_vector.h>
+
+
+#include "memory/tracking_memory_resource.hpp"
+#include "types.hpp"
+#include "utils/meta_tables/abstract_meta_table.hpp"
+
+namespace opossum {
+
+struct ResourceRecord {
+  OperatorType operator_type;
+  std::string operator_data_structure;
```

Comment:
Using strings in such a frequently accessed place might add a significant overhead.
I think it's okay as this is still in a "what is possible" stage, but for later on: is there any reason not to use an `enum class` for such data structures?